diff --git a/usr/lib/live/build/chroot_archives b/usr/lib/live/build/chroot_archives
index 952fbea22..1d8b9fec0 100755
--- a/usr/lib/live/build/chroot_archives
+++ b/usr/lib/live/build/chroot_archives
@@ -212,6 +212,9 @@ case "${_ACTION}" in
 				Echo_warning "Local packages must be named with suffix '_all.deb' or '_\$architecture.deb'."
 			fi
 		fi
+		
+		sed -i "s/^deb-src/#deb-src/g" "chroot/etc/apt/zz-sources.list"
+		sed -i 's!devel!./devel!' chroot/etc/apt/zz-sources.list
 
 		if Find_files chroot/root/packages/*.deb
 		then
diff --git a/usr/lib/live/build/source_iso b/usr/lib/live/build/source_iso
index 56aa48caf..6045022e6 100755
--- a/usr/lib/live/build/source_iso
+++ b/usr/lib/live/build/source_iso
@@ -105,6 +105,9 @@ cat > chroot/source.sh << EOF
 xorriso ${XORRISO_OPTIONS} -o source.iso source
 EOF
 
+sed -i '1d' chroot/etc/apt/sources.list
+echo 'deb [trusted=yes] file:/media/cdrom devel contrib main non-free' >> chroot/etc/apt/zz-sources.list
+
 Chroot chroot "sh source.sh"
 
 # Move image
